File: 1560.most-visited-sector-in-a-circular-track.cpp
/*
 * @lc app=leetcode id=1560 lang=cpp
 *
 * [1560] Most Visited Sector in  a Circular Track
 */
#include <vector>
#include <numeric>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<int> mostVisited(int n, vector<int> &rounds)
    {
        vector<int> res;
        res.reserve(n);
        if (rounds.back() >= rounds.front())
        {
            for (int i = rounds.front(); i != rounds.back() + 1; ++i)
            {
                res.push_back(i);
            }
            return res;
        }
        else
        {
            for (int i = 1; i != rounds.back() + 1; ++i)
            {
                res.push_back(i);
            }
            for (int i = rounds.front(); i != n + 1; ++i)
            {
                res.push_back(i);
            }
            return res;
        }
        return res;
    }
};
// @lc code=end

File: py/1560.most-visited-sector-in-a-circular-track.py
#
# @lc app=leetcode id=1560 lang=python3
#
# [1560] Most Visited Sector in  a Circular Track
#

# @lc code=start
class Solution:
    def mostVisited(self, n: int, rounds: List[int]) -> List[int]:
        if rounds[-1] >= rounds[0]:
            return list(range(rounds[0], rounds[-1]+1))
        else:
            return list(range(1, rounds[-1]+1))+list(range(rounds[0], n+1))
# @lc code=end
